linera-core/src/client.rs

```diff
@@ -2259,16 +2595,55 @@ where
     pub async fn publish_blob(
         &self,
         hashed_blob: HashedBlob,
-    ) -> Result<ClientOutcome<(BlobId, Certificate)>, ChainClientError> {
+    ) -> Result<ClientOutcome<Certificate>, ChainClientError> {
         self.client.local_node.cache_recent_blob(&hashed_blob).await;
         self.state_mut()
             .pending_blobs
             .insert(hashed_blob.id(), hashed_blob.clone());
         self.execute_operation(Operation::System(SystemOperation::PublishBlob {
             blob_id: hashed_blob.id(),
         }))
-        .await?
-        .try_map(|certificate| Ok((hashed_blob.id(), certificate)))
+        .await
+    }
+
+    /// Reads some blob. This is test-only, so we can test without a Wasm application.
+    #[cfg(with_testing)]
+    pub async fn read_blob(
+        &mut self,
+        blob_id: BlobId,
+    ) -> Result<ClientOutcome<Certificate>, ChainClientError> {
+        self.execute_operation(Operation::System(SystemOperation::ReadBlob { blob_id }))
+            .await
+    }
+
+    /// Publishes and reads some blobs. This is test-only, so we can test without a Wasm application.
+    #[cfg(with_testing)]
+    pub async fn read_and_publish_blobs(
+        &mut self,
+        blobs_to_read: &[BlobId],
+        blobs_to_publish: &[HashedBlob],
+    ) -> Result<ClientOutcome<Certificate>, ChainClientError> {
+        ensure!(
+            !blobs_to_read.is_empty() || !blobs_to_publish.is_empty(),
+            ChainClientError::NoBlobsToPublishOrReadProvided
```

Comment:
Even if we keep this method, let's not introduce a test-only error variant for it.

But I really think we should just remove it, and use `execute_operations` instead. (See other comment.)